### PR TITLE
feat: implement aggregate

### PR DESCRIPTION
**Pull request by voslartomas**
## Description

Adding `aggregatePaged` to be able to use advanced features of Mongo Atlas as autocomplete search.

Note: we don't need projections and populate here as it should be done via stagePipelines in aggregation.

### CODE REVIEW

**Comment by ExtraBB** [comment]
Looks good! Could you also update the `README.md` file? Then I'll merge it.

File: src/index.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-import { Schema, PopulateOptions } from "mongoose";
-import { generateCursorQuery, generateSort } from "./query";
+import { Schema, PopulateOptions, PipelineStage, Collection, Model } from "mongoose";
```
Comment (ExtraBB) [inline src/index.ts:1]: Remove unused  `Collection` import

File: src/index.ts [hunk 2]
```diff
@@ -64,8 +64,58 @@ export default function (schema: Schema, pluginOptions?: IPluginOptions) {
...
+            paginatedPipeline.push({ $limit: options.limit + 1 });
+        }
+    
+        // Execute the aggregate query
+        const cursor = mongoCollection.aggregate<T>(paginatedPipeline);
+    
+        return cursor;
+    }
+    
+    async function aggregatePaged<T>(
```
Comment (ExtraBB) [inline src/index.ts:93]: Looks good! Could you also update the README.md file?

**Review by ExtraBB — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (voslartomas) [reply src/index.ts:93]: All right, updated

**Pull request merged**